fe/fe-core/src/main/java/com/starrocks/ha/BDBStateChangeListener.java

```diff
@@ -70,7 +69,7 @@ public synchronized void stateChange(StateChangeEvent sce) throws RuntimeExcepti
             }
         }
         Preconditions.checkNotNull(newType);
-        GlobalStateMgr.getCurrentState().notifyNewFETypeTransfer(newType);
+        StateChangeExecutor.getInstance().notifyNewFETypeTransfer(newType);
```

Comment:
should create an interface that has `notifyNewFETypeTransfer`
Then we should setup into this Listener, then this listener can call-back one by one.

Reply:
in fact, this listener only does one thing, put new type into a queue, which is in executor, then when executor gets new type from the queue, it will call each module's s type change function.

Reply:
maybe you mean put an interface in executor like this?
```
newType = queue.top();
for (auto e : modules) {
  e.changeTo(newType);
}

Reply:
if this is multi-raft case, i think it's fine, but right now all modules in FE belongs to one BDB(one group), so i think better to make state change **serializable for all modules in one group**.
if multiple listeners, it means different modules might have different roles because they are handled by different state change executor threads, for example, GlobalStateMgr is leader but StarMgr is follower, this seems not acceptable.